src/main/java/dk/cngroup/trainings/spring/springassignment/service/helper/AnimalValidationService.java

```diff
@@ -0,0 +1,38 @@
+package dk.cngroup.trainings.spring.springassignment.service.helper;
+
+import dk.cngroup.trainings.spring.springassignment.model.Animal;
+import dk.cngroup.trainings.spring.springassignment.service.exception.InvalidAnimalException;
+
+public class AnimalValidationService {
+
+	public static void validate(Animal animal) throws InvalidAnimalException {
+		isAnimalNotNull(animal);
```

Comment:
the naming does not make much sense... usually when you call something `is...` then you expect that it returns `boolean` :). Also, now you have some duplicated code, because in every method, you are throwing the exception. It would be better if it returned `boolean` and then you would just check that all of them are true, otherwise you throw exception.

Reply:
if you really wanted it like this, you would need to name it maybe `checkThatAnimalIsNotNull`...

Reply:
Fixed. That was exactly when i switched to exceptions. I knew functionality won't change but I had to leave for the train :D